### PR TITLE
WD-2757- WebCLI history

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Add history to the WebCLI.

## QA

- Pull code
- Run `dotrun clean && dotrun serve`
- Open http://0.0.0.0:8036/
- Using a local controller go to a model details page.
- Using the CLI at the bottom enter a few commands.
- Press the up and down arrows and you should be able to navigate through your history.
- Refresh the page and you should still be able to navigate through your history.

## Details

https://warthogs.atlassian.net/browse/WD-2757
Fixes: #724.

## Screenshots

Screen Recording 2023-03-21 at 12 36 22 pm

### CODE REVIEW

File: src/components/WebCLI/WebCLI.tsx [hunk 1]
```diff
@@ -35,16 +43,53 @@ const WebCLI = ({
 }: Props) => {
   const [connection, setConnection] = useState<Connection | null>(null);
   const [shouldShowHelp, setShouldShowHelp] = useState(false);
+  const [historyPosition, setHistoryPosition] = useState(0);
   const inputRef = useRef<HTMLInputElement>(null);
   const wsMessageStore = useRef<string>("");
   let [output, setOutput] = useState("");
   const sendAnalytics = useAnalytics();
   const storeState = useStore().getState();
+  const [cliHistory, setCLIHistory] = useLocalStorage<string[]>(
```
Comment (goulinkh) [inline src/components/WebCLI/WebCLI.tsx:52]: We should consider adding a size limit to the size of history commands, where the size limit of the local storage is ~5-10 Mb which is huge however parsing and stringifying for example a 5Mb of JSON may not be ideal for performance.
having a limit around 200 items should be fine.

Comment (huwshimi) [reply src/components/WebCLI/WebCLI.tsx:54]: Good idea! I've updated this.

**Review by goulinkh — APPROVED** [review]
This works really well, I would just set the cursor at the end of the line if possible as sometimes it's at the beginning

**Comment by huwshimi** [comment]
> This works really well, I would just set the cursor at the end of the line if possible as sometimes it's at the beginning

Done!

**Pull request merged**